acme/problems.go

```diff
@@ -70,3 +72,19 @@ func AgreementRequiredProblem(detail string) *ProblemDetails {
 		HTTPStatus: http.StatusForbidden,
 	}
 }
+
+func ConnectionProblem(detail string) *ProblemDetails {
+	return &ProblemDetails{
+		Type:       connectionErr,
+		Detail:     detail,
+		HTTPStatus: http.StatusBadGateway, // TODO(@cpu): Change this later?
```

Comment:
I think it makes the most sense to leave out HTTPStatus for stuff that is returned in the error field of challenges.